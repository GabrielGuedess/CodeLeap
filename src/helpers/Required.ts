export type Required<T> = {
  [P in keyof T]-?: T[P];
};
