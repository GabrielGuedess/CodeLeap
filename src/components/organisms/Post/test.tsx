import { screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { renderWithTheme } from 'utils/tests/helpers';

import { Post } from '.';

const deletePostMock = jest.fn();
const editPostMock = jest.fn();

jest.mock('hooks/usePost', () => ({
  usePost: () => ({
    deletePost: deletePostMock,
    editPost: editPostMock,
  }),
}));

describe('<Post />', () => {
  it('should render the post correctly', async () => {
    // Arrange
    const { container } = renderWithTheme(
      <Post
        id="1"
        title="Title"
        username="GabrielGuedess"
        createdAt={new Date('2023-04-07T00:00:00.000Z')}
        content="Text"
      />,
    );

    // Assert
    await waitFor(() => {
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  it('should change info post when click on edit button', async () => {
    renderWithTheme(
      <Post
        id="1"
        title="Title"
        username="GabrielGuedess"
        createdAt={new Date('2023-04-07T00:00:00.000Z')}
        content="Text"
      />,
    );

    // Act
    await userEvent.click(screen.getByLabelText('Edit Button'));
    await userEvent.type(screen.getByLabelText('Title'), 'Gabriel');
    await userEvent.click(screen.getByRole('button', { name: 'Save' }));

    // Assert
    await waitFor(() => {
      expect(editPostMock).toHaveBeenCalledTimes(1);
    });
  });

  it('should delete post when click on delete button', async () => {
    renderWithTheme(
      <Post
        id="1"
        title="Title"
        username="GabrielGuedess"
        createdAt={new Date('2023-04-07T00:00:00.000Z')}
        content="Text"
      />,
    );

    // Act
    await userEvent.click(screen.getByLabelText('Delete Button'));
    await userEvent.click(screen.getByLabelText('Delete'));

    // Assert
    await waitFor(() => {
      expect(deletePostMock).toHaveBeenCalled();
    });
  });
});
