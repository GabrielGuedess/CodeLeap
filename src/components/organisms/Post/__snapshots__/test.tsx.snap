// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Post /> should render the post correctly 1`] = `
.c5 {
  cursor: pointer;
  color: #FFFFFF;
}

.c2 {
  font-size: 2.2rem;
  font-weight: 700;
  color: #FFFFFF;
}

.c4 {
  background: transparent;
  border: none;
}

.c4:focus {
  box-shadow: #B8BEF3 0px 0px 0px 1.5px;
}

.c0 {
  overflow: hidden;
  background: #FFFFFF;
  border: 1px solid #CCCCCC;
  border-radius: 1.6rem;
  width: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
  padding: 2.7rem 3.7rem;
  background: #6084e9;
}

.c1 h2 {
  overflow: hidden;
  max-width: 80%;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 2.38rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 2.4rem;
  width: 100%;
  padding: 2.4rem;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-bottom: 1.6rem;
}

.c8 {
  overflow: hidden;
  max-width: 50%;
  font-size: 1.3rem;
  font-weight: 700;
  color: #000;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.c9 {
  font-size: 1.3rem;
  color: #666666;
}

.c10 {
  font-size: 1.4rem;
}

@media (max-width:450px) {
  .c7 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    gap: 1rem;
  }
}

@media (max-width:450px) {
  .c8 {
    max-width: 90%;
  }
}

@media (min-width:450px) {
  .c8 {
    font-size: 1.4rem;
  }
}

@media (min-width:768px) {
  .c8 {
    font-size: 1.8rem;
  }
}

@media (min-width:450px) {
  .c9 {
    font-size: 1.4rem;
  }
}

@media (min-width:768px) {
  .c9 {
    font-size: 1.8rem;
  }
}

@media (min-width:768px) {
  .c10 {
    font-size: 1.8rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <h2
      class="c2"
      color="secondary"
    >
      Title
    </h2>
    <div
      class="c3"
    >
      <button
        aria-controls="radix-:r0:"
        aria-expanded="false"
        aria-haspopup="dialog"
        class="c4"
        data-state="closed"
        type="button"
      >
        <div
          class="c5"
        >
          <svg
            aria-label="Delete Button"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zm2.46-7.12l1.41-1.41L12 12.59l2.12-2.12l1.41 1.41L13.41 14l2.12 2.12l-1.41 1.41L12 15.41l-2.12 2.12l-1.41-1.41L10.59 14l-2.13-2.12zM15.5 4l-1-1h-5l-1 1H5v2h14V4z"
              fill="currentColor"
            />
          </svg>
        </div>
      </button>
      <button
        aria-controls="radix-:r3:"
        aria-expanded="false"
        aria-haspopup="dialog"
        class="c4"
        data-state="closed"
        type="button"
      >
        <div
          class="c5"
        >
          <svg
            aria-label="Edit Button"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m7 17.013l4.413-.015l9.632-9.54c.378-.378.586-.88.586-1.414s-.208-1.036-.586-1.414l-1.586-1.586c-.756-.756-2.075-.752-2.825-.003L7 12.583v4.43zM18.045 4.458l1.589 1.583l-1.597 1.582l-1.586-1.585l1.594-1.58zM9 13.417l6.03-5.973l1.586 1.586l-6.029 5.971L9 15.006v-1.589z"
              fill="currentColor"
            />
            <path
              d="M5 21h14c1.103 0 2-.897 2-2v-8.668l-2 2V19H8.158c-.026 0-.053.01-.079.01c-.033 0-.066-.009-.1-.01H5V5h6.847l2-2H5c-1.103 0-2 .897-2 2v14c0 1.103.897 2 2 2z"
              fill="currentColor"
            />
          </svg>
        </div>
      </button>
    </div>
  </div>
  <div
    class="c6"
  >
    <div
      class="c7"
    >
      <strong
        class="c8"
      >
        @
        GabrielGuedess
      </strong>
      <span
        class="c9"
      >
        3 days ago
      </span>
    </div>
    <span
      class="c10"
    >
      Text
    </span>
  </div>
</div>
`;
